### PR TITLE
chore: allow checking http fullnodes

### PR DESCRIPTION
**Pull request by luislhl**
### Acceptance Criteria
- We should be able to make the daemon check fullnodes through `http` instead of `https`. Each fullnode should define its protocol, instead of having `https` hard-coded.

### Security Checklist
- [ ] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

**Review by r4mmer — APPROVED** [review]

File: gateways/clients/hathor_core_client.py [hunk 1]
```diff
@@ -33,14 +33,15 @@
...
     DEFAULT_TIMEOUT = 60  # seconds
 
-    def __init__(self, domain: Optional[str] = None) -> None:
+    def __init__(self, url: Optional[str] = None, prefix: str = "https://") -> None:
         """Client to make async requests
 
-        :param domain: domain where the requests will be made, defaults to config `hathor_core_domain`
-        :type domain: str, optional
+        :param url: url where the requests will be made, defaults to config `hathor_core_url`
+        :type url: str, optional
```
Comment (tuliomir) [inline gateways/clients/hathor_core_client.py:40]: suggestion: Add the `prefix` parameter to the docstrings

File: gateways/clients/hathor_core_client.py [hunk 2]
```diff
@@ -114,13 +115,14 @@ async def post(
 class HathorCoreClient:
     """Client to make requests
 
-    :param domain: domain where the requests will be made, defaults to config `hathor_core_domain`
-    :type domain: str, optional
+    :param url: url where the requests will be made, defaults to config `hathor_core_url`
+    :type url: str, optional
```
Comment (tuliomir) [inline gateways/clients/hathor_core_client.py:119]: suggestion: Add the `use_ssl` parameter to the docstrings

File: gateways/clients/hathor_core_client.py [hunk 3]
```diff
@@ -114,13 +115,14 @@ async def post(
...
     """
 
-    def __init__(self, domain: Optional[str] = None) -> None:
-        self.domain = domain or HATHOR_CORE_DOMAIN
+    def __init__(self, url: Optional[str] = None, use_ssl: bool = True) -> None:
+        self.url = url or HATHOR_CORE_URL
         self.log = logger.new(client="sync")
+        self.protocol = "https" if use_ssl else "http"
```
Comment (tuliomir) [inline gateways/clients/hathor_core_client.py:125]: question: Should `protocol` be a private property? If so, I didn't see any use of it, which may be a bug.
If it's a public property, maybe add it to some docstrings as well.

File: tests/unit/gateways/clients/test_hathor_core_client.py
Comment (tuliomir) [inline tests/unit/gateways/clients/test_hathor_core_client.py:1]: suggestion: The tests do not interact with the newly added parameters `prefix` and `use_ssl`. I'd suggest adding new tests to cover these, as they may help identify solve eventual migration issues we face in this refactor.

File: gateways/clients/hathor_core_client.py (hunk 1 above)
Comment (luislhl) [reply gateways/clients/hathor_core_client.py:40]: Actually this is a leftover, I've removed it: https://github.com/HathorNetwork/hathor-explorer-service/pull/326/commits/71588c0930c6dae82d77c50a2e24e7552ff3edcd

File: gateways/clients/hathor_core_client.py (hunk 2 above)
Comment (luislhl) [reply gateways/clients/hathor_core_client.py:118]: Actually this is a leftover, I've removed it: https://github.com/HathorNetwork/hathor-explorer-service/pull/326/commits/71588c0930c6dae82d77c50a2e24e7552ff3edcd

File: gateways/clients/hathor_core_client.py (hunk 3 above)
Comment (luislhl) [reply gateways/clients/hathor_core_client.py]: Actually this is a leftover, I've removed it: https://github.com/HathorNetwork/hathor-explorer-service/pull/326/commits/71588c0930c6dae82d77c50a2e24e7552ff3edcd

File: tests/unit/gateways/clients/test_hathor_core_client.py
Comment (luislhl) [reply tests/unit/gateways/clients/test_hathor_core_client.py:1]: Actually this is a leftover, I've removed it: https://github.com/HathorNetwork/hathor-explorer-service/pull/326/commits/71588c0930c6dae82d77c50a2e24e7552ff3edcd

**Review by tuliomir — APPROVED** [review]

**Pull request merged**